vars/buildGradlePlugin.groovy

```diff
@@ -96,7 +97,7 @@ def call(Map config = [:]) {
                   if (config.coverallsToken) {
                     tasks += " coveralls"
                   }
-                  if(config.sonarToken) {
+                  if(config.sonarToken && (BRANCH_NAME in ["master", "main"] || params.RUN_SONAR)) {
```

Comment:
Could we make a config value out of `['master', 'main']` and use regex match?
You could add something down the line after line 23

```groovy
config.sonarQubeBranchPattern = config.sonarQubeBranchPattern ?: "(master|main)"
```

and then do a match against this pattern in the if block. This would allow the jenkinsfile to adjust the pattern.

Reply:
This sounds good. I'll do that.

Reply:
Done.